PowerBoardSoftware/PowerBoardSoftware.ino

```diff
@@ -0,0 +1,430 @@
+const uint16_t MISC_30V_CURRENT_READING   = 3107; //12V Board, Rockets, Auxiliary
+const uint16_t VAC_CURRENT_READING        = 3108; //Vacuum
+
+//Bus control
+//Disable = 0, Enable = 1
+const uint16_t MOTOR_ENABLE      = 3100; //M1,M2,M3,M4,M5,M6,M7,Spare (last 2 unused atm)
+const uint16_t ACT_LOG_ENABLE    = 3101; //Actuation,Logic
+const uint16_t MISC_30V_ENABLE   = 3102; //12V Board, Rockets, Auxiliary
+const uint16_t VAC_ENABLE        = 3103; //Vacuum
+const uint16_t PATCH_ENABLE      = 3104; //PB1,PB2,PB3,PB4,PB5,PB6,PB7,PB8
+
+
+//TIVA PIN MAPPINGS
+//ANALOG PINS
+//30V
+#define RKT_ISENSE PE_4
+#define TWV_ISENSE PE_5
+#define AUX_ISENSE PD_3
+#define VAC_ISENSE PE_2
+#define M1_ISENSE PD_4
+#define M2_ISENSE PK_1
+#define M3_ISENSE PK_3
+#define M4_ISENSE PD_2
+#define M5_ISENSE PB_4
+#define M6_ISENSE PE_1
+//12V
+#define LOG_ISENSE PD_7
+#define ACT_ISENSE PE_3
+
+//TEMPORARY MAPPINGS BECAUSE I DID BAD
+#define LOG_EN PA_6
+#define ACT_EN PD_7
+//end temp stuff
+
+//DIGITAL PINS
+//Error LED
+#define ERROR PB_3
+//30V
+#define RKT_EN PB_2
+#define TWV_EN PC_6
+#define AUX_EN PC_7 
+#define VAC_EN PC_5
+#define M1_EN PP_1
+#define M2_EN PD_5
+#define M3_EN PQ_0
+#define M4_EN PN_5
+#define M5_EN PP_0
+#define M6_EN PE_0
+//12V
+//#define LOG_EN PM_4
+//#define ACT_EN PA_6
+#define VOUT_EN PM_5
+//Patch Panel
+#define PB1_EN PL_3
+#define PB2_EN PF_3
+#define PB3_EN PL_2
+#define PB4_EN PG_0
+#define PB5_EN PL_1
+#define PB6_EN PL_4
+#define PB7_EN PL_0
+#define PB8_EN PL_5
+
+//Current sense mapping values
+const uint16_t ADCMin = 330;
+const uint16_t ADCMax = 4095;
+const uint16_t ISENSEMin = 0;
+const uint16_t ISENSEMax = 40000;
+const float CURR_SCALER = 1.15;
+
+//Overcurrent constants (amps)
+const uint8_t OCP_MOTOR = 180; //temp fix due to inaccurate current readings on Motor busses
+const uint8_t OCP_12V = 18;
+const uint8_t OCP_30V = 18;
+const uint8_t OCP_RKT = 4;
+const uint8_t OCP_VAC = 13;
+const uint16_t RESET_TIME = 5000;  //ms between tripping OCP and turn on
+
+
+struct Bus
+{
+  int EN_PIN;
+  int ISENSE_PIN = -1;//default for busses without ISENSE pins
+  bool enabled = false;
+
+  Bus(const int en, const int isense)
+  {
+    EN_PIN = en;
+    ISENSE_PIN = isense;
+    pinMode(ISENSE_PIN,INPUT);
+    pinMode(EN_PIN,OUTPUT);
+  }
+
+  Bus(const int en)
+  {
+    EN_PIN = en;
+    pinMode(EN_PIN,OUTPUT);
+  }
+
+};
+
+Bus busses_Motor[6] = {Bus(M1_EN,M1_ISENSE),Bus(M2_EN,M2_ISENSE),Bus(M3_EN,M3_ISENSE),Bus(M4_EN,M4_ISENSE),Bus(M5_EN,M5_ISENSE),Bus(M6_EN,M6_ISENSE)};
+Bus busses_12V[3] = {Bus(ACT_EN,ACT_ISENSE),Bus(LOG_EN,LOG_ISENSE),Bus(VOUT_EN)};
+Bus busses_30V[3] = {Bus(TWV_EN,TWV_ISENSE),Bus(RKT_EN,RKT_ISENSE),Bus(AUX_EN,AUX_ISENSE)};
+Bus vac = Bus(VAC_EN,VAC_ISENSE);
+//the last time in millis() when telemetry was sent
+unsigned long lastTelemTime = 0;
+
+
+void setup() {
+  Serial.begin(9600);
+  RoveComm.begin(RC_POWERBOARD_FOURTHOCTET, RC_ROVECOMM_ETHERNET_POWERBOARD_PORT);
```

Comment:
Will be passing in EthernetServer instance with designated port (see RoveCommTCP example for details)